src/mixins/filing-template-mixin.ts

```diff
@@ -169,9 +173,41 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
           agreementType: this.getAgreementType
         }
       },
-      // *** FUTURE: implement these:
-      changeofRegistration: undefined,
-      registration: undefined
+      changeofRegistration: !isChangeReg ? undefined : {
+        business: {
+          identifier: this.getBusinessId,
+          naics: this.getCurrentNaics
+        },
+        offices: this.getOfficeAddresses,
+        contactPoint: {
+          email: this.getBusinessContact.email,
+          phone: this.getBusinessContact.phone,
+          ...this.getBusinessContact.extension
+            ? { extension: +this.getBusinessContact.extension }
+            : {}
+        },
+        parties: this.getPeopleAndRoles
+      },
+      registration: !isRegistration ? undefined : {
+        business: {
+          identifier: this.getBusinessId,
+          naics: this.getCurrentNaics
+        },
+        contactPoint: {
+          email: this.getBusinessContact.email,
+          phone: this.getBusinessContact.phone,
+          ...this.getBusinessContact.extension
+            ? { extension: +this.getBusinessContact.extension }
+            : {}
+        },
+        nameRequest: {
+          legalType: this.getEntityType,
+          legalName: this.getApprovedName,
+          nrNumber: this.getNameRequestNumber
+        },
+        offices: this.getOfficeAddresses,
+        parties: this.getPeopleAndRoles
+      }
```

Comment:
This is OK for now. I know Lewis is also working on this.

Reply:
I was talking with Lewis before making to many changes I wanted to talk with him more.